src/application/common/logger.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as Logger from 'bunyan';
+
+const ILogger = Symbol("ILogger");
+
+interface ILogger {
+    addStream(stream: Logger.Stream): Promise<void>;
```

Comment:
1. We can it's just more logic, but OK I do think it's useful for extensions or components so let's do it.

Re token: Yes so params: any[] no ... 

About 2, yes good point I do think it would be better to have a notification then, prohibiting the usage would complicate things